_posts/wiki/2019-09-10-Jekyll-Installation-Tutorial.md

```diff
@@ -0,0 +1,96 @@
+
+# What is Jekyll?
+
+Jekyll is a static site generator which helps to organize the website data for
+the HIRO Group and generate the website. By installing Jekyll you will be able
+to host the website locally to see how your article or added page looks before
+pushing changes to the HIRO Group repository.
+
+# Steps For Installing Jekyll
+
+ 1. Check Ruby Version: Ruby version must be updated to 2.4.0 or above if not see Installing RVM & Updating Ruby below. Use: `ruby -v`
+ 2. Check to see if you have Ruby Gems: `gems -v`
+ 3. Install Dependencies: `sudo apt install ruby-full build-essential zlib1g-dev`
+ 4. Configure Installation Path for Gems. The following will add the appropriate paths to the `.bashrc` file:
+    ~~~bash
+    echo '# Install Ruby Gems to ~/gems' >> ~/.bashrc
+    echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc
+    echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
+    source ~/.bashrc
+    ~~~
+ 5. Install Jekyll:`gem install jekyll bundler`
+ 6. Generate a Gem File
+  * Either create a new file in a text editor name it “Gemfile” with no extension. Save it in your local repository
+  * Or, while in your local repository enter in terminal: `bundle init`, this will generate a new gem file. Delete everything inside of it and move on to the next step.
+  * In your newly generate gem file add the following:
+    ~~~ruby
+    source "https://rubygems.org"
+    gem "github-pages", group: :jekyll_plugins
+    ~~~
+
+# Installing RVM & Updating Ruby
+
+Ruby version manager (RVM) allows for users to work on legacy code using older
+versions of ruby by enabling easy switching between versions. This allows us to
+update our ruby version more easily.
+
+1. Download RVM
+ * Run the following in your bash terminal:
+    ~~~bash
+    curl -sSL http://get.rvm.io | sudo bash -s stable -- --ignore-dotfiles
+    ~~~
+ * Check if the user group rvm has been added:
+    ~~~bash
+    groups
+    getent group rvm
+    ~~~
+ * Add yourself to the rvm group:
+    ~~~bash
+    sudo usermod -a -G rvm <your username here>
+    ~~~
+ * In your terminal enter `getent group rvm` and  you should see: `rvm:x :###:<your username here>`
+ * Log out then log in to activate changes.
+2. Source your terminal to access the newer versions of Ruby: add the following to your bashrc file: `source /etc/profile.d/rvm.sh`
+3. Update Ruby:
+  * See what ruby versions are available: `rvm list known`
+  * Select any version 2.4.0 or above (install a fairly new version)
+  * This command takes care of dependencies:
+    ~~~bash
+    sudo apt install libyaml-devel readline-devel libffi-devel sqlite-devel
+    ~~~
+  * Finally install updated Ruby version
+    ~~~bash
+    rvm install ruby-<version number here ex: 2.6.3>
+    ~~~
+  * Check Ruby version: `ruby -v`
+  * Finished!
+
+# Next Steps
+
+1. Locally host the website:
+* This command compiles the files (only do this the first time you host): `bundle exec jekyll serve`
```

Comment:
Do you have a preference on how group members set Jekyll up? I haven't had any problems the way I installed with bundler. I can change it if you would like.

Reply:
So bundler is basically avoiding the need of installing missing gems yourself. It looks at the `_config.yml`, it sees that it needs some gems to run (namely, `jekyll-sitemap, rouge, jekyll-seo-tag, jemoji`), and it downloads them for you.
I don't honestly care which one is better, maybe bundler smoothes out the process a little bit? If so, it'd be worth adding a `gemfile` to the repo already so that you don't have to create it manually.

Reply:
I'll go ahead and add a gemfile to this pull request and edit the guide.